cmd/kube-apiserver/app/server.go

```diff
@@ -657,7 +657,15 @@ func buildServiceResolver(enabledAggregatorRouting bool, hostname string, inform
 	if localHost, err := url.Parse(hostname); err == nil {
 		serviceResolver = aggregatorapiserver.NewLoopbackServiceResolver(serviceResolver, localHost)
 	}
-	return serviceResolver
+	return &MutableServiceResolver{Delegate: serviceResolver}
```

Comment:
is there a reason we can't just override aggregator.ServiceResolver in MutateAggregatorFunc? I'd rather not leak test-only requirements into this setup

Reply:
@liggitt overriding does not work because that variable is captured at multiple other parts of the stack.  So overriding in just `MutateAggregatorFunc` is not sufficient since other parts of the code would not notice the change.  I am not a fan of this change either for the same reason as you mentioned but I could not think of another way to make it work...

Reply:
oh, blegh

Reply:
another possibility is to make the call to buildServiceResolver overrideable for tests (see `capabilities.SetForTests` as an example), swap the entire impl prior to the test run, and restore it afterwards

Reply:
that way the runtime behavior of non-test servers doesn't have any weirdness mixed in

Reply:
@liggitt see if you like the updated approach better.